tools/depcheck/pkg/cmd/trace/pkg.go

```diff
@@ -0,0 +1,131 @@
+package trace
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/gonum/graph"
+	"github.com/gonum/graph/concrete"
+
+	depgraph "github.com/openshift/origin/tools/depcheck/pkg/graph"
+)
+
+var (
+	// lets us distinguish core library deps
+	knownVendorRoots = []string{
+		"/vendor/bitbucket.org/",
+		"/vendor/cloud.google.com",
+		"/vendor/github.com",
+		"/vendor/go4.org",
+		"/vendor/golang.org",
+		"/vendor/google.golang.org",
+		"/vendor/gopkg.in",
+		"/vendor/k8s.io/",
+		"/vendor/vbom.ml",
+	}
+	baseRepo = "github.com/openshift/origin"
+)
+
+type Package struct {
+	ImportPath  string
+	Imports     []string
+	TestImports []string
+}
+
+type PackageList struct {
+	Packages []Package
+}
+
+// removeCoreLibraryImports receives a Package and filters any
+// dependencies not part of the base repo or its vendor tree
+func (p *PackageList) removeCoreLibraryImports(imports []string) []string {
+	validImports := []string{}
+	for _, dep := range imports {
+		if !strings.Contains(dep, "/vendor/") && !strings.HasPrefix(dep, baseRepo) {
+			continue
+		}
+
+		validImports = append(validImports, dep)
+	}
+
+	return validImports
+}
+
+// TODO: provide a way for consumers of this command
+// to specify a differentiation between vendored and
+// non-vendored dependencies
+func (p *PackageList) FilteredAdd(pkg Package, excludes []string) {
+	for _, exclude := range excludes {
+		if strings.Contains(pkg.ImportPath, exclude) {
+			return
+		}
+	}
+
+	if !strings.Contains(pkg.ImportPath, "/vendor/") {
+		pkg.Imports = p.removeCoreLibraryImports(pkg.Imports)
+		pkg.TestImports = p.removeCoreLibraryImports(pkg.TestImports)
+		p.Add(pkg)
+		return
+	}
+
+	for _, valid := range knownVendorRoots {
+		if strings.Contains(pkg.ImportPath, valid) {
+			pkg.Imports = p.removeCoreLibraryImports(pkg.Imports)
+			pkg.TestImports = p.removeCoreLibraryImports(pkg.TestImports)
+			p.Add(pkg)
+			return
+		}
+	}
+}
+
+func (p *PackageList) Add(pkg Package) {
+	p.Packages = append(p.Packages, pkg)
+}
+
+func BuildGraph(packages *PackageList) (graph.Directed, error) {
```

Comment:
doc what is missing from this graph.  You will be snipping core library dependencies.